x-pack/plugins/synthetics/server/saved_objects/synthetics_monitor.ts

```diff
@@ -27,6 +73,25 @@ export const LEGACY_SYNTHETICS_MONITOR_ENCRYPTED_TYPE = {
      * payload on the `secrets` key. This ensures performant decryption. */
     ...secretKeys,
   ]),
+  attributesToIncludeInAAD: new Set([
+    LegacyConfigKey.SOURCE_ZIP_URL,
+    LegacyConfigKey.SOURCE_ZIP_USERNAME,
+    LegacyConfigKey.SOURCE_ZIP_PASSWORD,
+    LegacyConfigKey.SOURCE_ZIP_FOLDER,
+    LegacyConfigKey.SOURCE_ZIP_PROXY_URL,
+    LegacyConfigKey.ZIP_URL_TLS_CERTIFICATE_AUTHORITIES,
+    LegacyConfigKey.ZIP_URL_TLS_CERTIFICATE,
+    LegacyConfigKey.ZIP_URL_TLS_KEY,
+    LegacyConfigKey.ZIP_URL_TLS_KEY_PASSPHRASE,
+    LegacyConfigKey.ZIP_URL_TLS_VERIFICATION_MODE,
+    LegacyConfigKey.ZIP_URL_TLS_VERSION,
+    LegacyConfigKey.THROTTLING_CONFIG,
+    LegacyConfigKey.IS_THROTTLING_ENABLED,
+    LegacyConfigKey.DOWNLOAD_SPEED,
+    LegacyConfigKey.UPLOAD_SPEED,
+    LegacyConfigKey.LATENCY,
+    ...attributesToIncludeInAAD,
+  ]),
```

Comment:
For the legacy synthetics monitor type, there were no excluded attributes, so I have added all of the defined non-secret legacy config keys, as well as all of the current definitions included attributes.

Reply:
Should we add some guidance to the docs once this merges?

Reply:
Yeah, I think it'd be great! The topic isn't trivial and having docs to forward devs to would be immensely helpful.

Reply:
https://github.com/elastic/kibana/issues/176097